jgrapht-core/src/main/java/org/jgrapht/alg/interval/KorteMoehringIntervalGraphRecognizer.java

```diff
@@ -339,96 +324,175 @@ public boolean isIntervalGraph()
         // TODO implement
         return null;
     }
-    
-    
-    
-    
-    
-    
-    
-    
-    
-    
-    private abstract class MPQNode
-    {
-        MPQNode left;
-        MPQNode right;
-        MPQNode parent;
-        
-        MPQNodeSetElement bag;
-        
-        MPQNode(MPQNodeSetElement bag) {
-            this.bag = bag;
+
+    /**
+     * A node of a modified PQ-tree
+     */
+    private abstract class MPQTreeNode {
+
+        /**
+         * The parent of the current node
+         */
+        MPQTreeNode parent;
+
+        /**
+         * Both P node and Q node must keep track of the left most and right most child
+         */
+        MPQTreeNode leftmostChild;
```

Comment:
this is wrong. Only the Q nodes does this. The P Nodes have just a pointer to one child.

Reply:
The Parent P node maintains a pointer to each of it's children and the children also have a pointer to the parent.